### PR TITLE
Fix initialization of Hijing interface

### PR DESCRIPTION
**Pull request by klendathu2k**
A cut-and-paste error was introduced when we added lazy initialization to the StarHijing interface ( Issue #4 ).  Resulting in incorrect initialization of the beams for hijing.  This PR resolves the issue.  On initialization of a dAu run, hijing now reports the correct beam parameters:

          ****************************************                                                                          **********
          *                                                *
          *         HIJING has been initialized at         *
          *             Ecm =     200.00 GeV/n             *
          *                                                *
          *        for A   (  2,  1) + A   (197, 79)       *
          **************************************************

### CODE REVIEW

**Comment by genevb** [comment]
> @genevb FYI -- once merged, we will need to patch libraries from SL20d on.

Thanks for the note, @klendathu2k . None of SL20d, SL21a, nor SL21b were used in any physics productions. So if I understand this correctly, only SL21c will need updated once merged. Too bad we didn't have this last week when we re-tagged SL21c. There's the possibility to update SL21c when there's a scheduled HPSS downtime around Oct. 19th if this can be ready by then.

**Review by genevb — COMMENTED** [review]
Looks like the _ablue issue was fixed, a couple functions to return a reference pointer to pythia, and sneaking in a path change too? So it's a little bit more than just fixing the issue, but I don't have any particular objections to updating SL21c with this.

File: StRoot/StarGenerator/Pythia8_2_35/StarPythia8.cxx [hunk 1]
```diff
@@ -30,7 +30,7 @@ StarPythia8::StarPythia8(const char *name) : StarGenerator(name)
   // auxilliary files required by pythia. 
   // NOTE:  When adding new versions of Pythia8, we need to specify
   // the version of the code in the source code
-  TString path = "StRoot/StarGenerator/"; path+= Pythia8_version; path+="/xmldoc/"; 
+  TString path = "StRoot/StarGenerator/"; path+= Pythia8_version; path+="/share/xmldoc/";
```
Comment (plexoos) [inline StRoot/StarGenerator/Pythia8_2_35/StarPythia8.cxx:33]: Wasn't this fixed in #3 already? What am I missing?

Comment (plexoos) [reply StRoot/StarGenerator/Pythia8_2_35/StarPythia8.cxx]: I merged the main branch into your PR branch and this change does not show up anymore. An easy fix.

**Review by plexoos — APPROVED** [review]

**Comment by genevb** [comment]
OK, I think we're agreed on tagging SL21c_5 in both repositories now. Also, it's still possible to apply the SL21c_1 through SL21c_4 tags to the version of star-mcgen codes that is in SL21c, which I would welcome you to do.

**Pull request merged**